c2corg_ui/static/js/suggestiondirective.js

```diff
@@ -0,0 +1,54 @@
+goog.provide('app.suggestionDirective');
+
+/**
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {angular.$sce} $sce Angular Strict Contextual Escaping service.
+ * @return {angular.Directive} Directive Definition Object.
+ * @ngInject
+ */
+app.suggestionDirective = function($compile, $sce) {
+  var html = '';
+  var header = '<span class="suggestion-text">' +
+                            '<span class="suggestion-title">' +
+                              '<span ng-bind-html="highlight(label, query)"></span>' +
+                              '<span ng-if="date_end">&nbsp;({{::date_end }})</span>' +
+                            '</span><br>' +
+                            '<span class="suggestion-info">' +
+                            '  <span ng-if="elevation">{{::elevation}}m</span>' +
+                            '</span>';
+  var footer =   '</span>';
+
+  var template = function(doc) {
+    var doctype = doc['doctype'];
+    if (doctype === 'waypoints') {
+      html = header + '</span><span class="icon-container"><span class="icon-{{::waypoint_type}}"></span>' + footer;
+    } else if (doctype === 'routes' || doctype === 'outings') {
+      var activitiesHtml = '';
+      for (var i = 0; i < doc['activities'].length; i++) {
+        activitiesHtml += '<span class="icon-' + doc['activities'][i] + '"></span>';
+      }
+      html = header + '</span><span class="icon-container">' + activitiesHtml + footer;
+    }
+    return html;
+  };
+
+  return {
+    restrict: 'E',
+    scope: {'doc': '@', 'label' : '@'}, // I left label because there were problems with escaping quotes in the title with json.parse
```

Comment:
I see :)
But your comment was confusing. Do we really need this comment anyway?